### DISCUSSION TITLE
Display a PDF file in Flet

### DISCUSSION THREAD

**Discussion by Lafalletes** [flet-dev/flet#310] [General]
Hi!

Is it possible to view a PDF file in Flet?

Thanks.

**Comment by pixel-studio-official** [comment]
i dont know

**Comment by ap4499** [comment]
Not currently I don't believe. 

I'm unsure if this would work for your purposes.. but you could (in code), convert to images, and then serve up those images of the PDF in the Flet GUI (https://flet.dev/docs/controls/image/)

**Comment by efermon** [comment]
This code may be useful for you:

import flet as ft
import os

def main(page: ft.Page):

    page.title = 'Read PDF'


    def open_file(e):
       if os.name == 'nt':
          # windows
          os.system(f"start {fic_pdf.value}")  
       else:
          # posix
          os.system(f"open {fic_pdf.value}")     
   
    fic_pdf = ft.TextField('/home/emilio/PDF/Introduction_flet.pdf', width=500)

    page.add(

      ft.Row(
          [fic_pdf,
            ft.IconButton(icon=ft.icons.TASK, icon_size=50, tooltip='Open PDF', on_click=open_file),
          ]
      )

    )


ft.app(target=main)

**Comment by mostafa-kessassi** [comment]
Yes you can now
use this
`pip install flet-pdfview`
I made it , I hope that will help
